sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Filtering/DocumentStream.cs

```diff
@@ -0,0 +1,194 @@
+        {
+            return DocumentStream.CheckFilters(this.dependencyFilterGroups, document, out errors);
+        }
+
+        public bool CheckFilters(ExceptionDocument document, out CollectionConfigurationError[] errors)
+        {
+            return DocumentStream.CheckFilters(this.exceptionFilterGroups, document, out errors);
+        }
+
+        public bool CheckFilters(Trace document, out CollectionConfigurationError[] errors)
+        {
+            return DocumentStream.CheckFilters(this.traceFilterGroups, document, out errors);
+        }
+
+        private static bool CheckFilters<TTelemetry>(
+            List<FilterConjunctionGroup<TTelemetry>> filterGroups,
+            TTelemetry document,
+            out CollectionConfigurationError[] errors)
+        {
+            var errorList = new List<CollectionConfigurationError>();
+            bool leastOneConjunctionGroupPassed = false;
+
+            if (filterGroups.Count == 0)
+            {
+                errors = Array.Empty<CollectionConfigurationError>();
+
+                // no filters for the telemetry type - filter out, we're not interested
+                return false;
+            }
+
+            // iterate over filter groups (filters within each group are evaluated as AND, the groups are evaluated as OR)
+            foreach (FilterConjunctionGroup<TTelemetry> conjunctionFilterGroup in filterGroups)
+            {
+                if (DocumentStream.CheckFiltersGeneric(document, conjunctionFilterGroup, errorList))
+                {
+                    // no need to check remaining groups, one OR-connected group has passed
+                    leastOneConjunctionGroupPassed = true;
+                    break;
+                }
+            }
+
+            errors = errorList.ToArray();
+
+            return leastOneConjunctionGroupPassed;
+        }
+
+        private static bool CheckFiltersGeneric<TTelemetry>(TTelemetry document, FilterConjunctionGroup<TTelemetry> filterGroup, List<CollectionConfigurationError> errorList)
+        {
+            bool filterPassed = false;
+
+            try
+            {
+                if (filterGroup.CheckFilters(document, out CollectionConfigurationError[] groupErrors))
+                {
+                    errorList.AddRange(groupErrors);
+                    filterPassed = true;
+                }
+            }
+            catch (System.Exception)
+            {
+                // the filters have failed to run (possibly incompatible field value in telemetry), consider the telemetry item filtered out by this conjunction group
+                ////!!!
+                ////errorList.Add(
+                ////    CollectionConfigurationError.CreateError(
+                ////        CollectionConfigurationErrorType.DocumentStreamFilterFailureToRun,
+                ////        string.Format(CultureInfo.InvariantCulture, "Document stream filter failed to run"),
+                ////        e));
+            }
```

Comment:
Did you comment this out or is this as-is from the AI SDK?

I don't see any TODOs here so I don't know what the expectation is.
(ie: should we log internally, should we transmit errors to LiveMetrics?)